code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -157,14 +159,16 @@
 			else
 				to_chat(user, "<span class='notice'>\The [src] doesn't seem to respect your authority.</span>")
 	else if(istype(W, /obj/item/kitchen/knife) && user.a_intent != INTENT_HARM)
-		to_chat(user, "<span class='notice'>You start attaching the [W] to \the [src]...</span>")
-		if(do_after(user, 40, target = src))
+		to_chat(user, "<span class='notice'>You start attaching \the [W] to \the [src]...</span>")
+		if(do_after(user, 25, target = src))
 			deputize(W, user)
 	else
 		return ..()
 
 /mob/living/simple_animal/bot/cleanbot/emag_act(mob/user)
 	..()
+	if(weapon)
```

Comment:
Note that if a silicon hacks an armed cleanbot, the weapon force will not be affected, as this does not called `emag_act()` and can instead be found in the parent `Topic()` proc. However, if someone deputizes the cleanbot while in the hacked state, the weapon will have full force due to the emag check above. Sorry to burden you with this.

By the way, the `emagged` var has two levels. 1 means the ID lock has been removed, 2 means the bot has been fully emagged and will begin hostile actions. Although, you might want to have a full power stabby cleanbot puttering about without it making obvious signs of being emagged.

Reply:
blah good catch, I'm on it